sound/soc/sof/core.c

```diff
@@ -418,8 +418,14 @@ static int sof_init_environment(struct snd_sof_dev *sdev)
 static int sof_probe_continue(struct snd_sof_dev *sdev)
 {
 	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct sof_dsp_power_state state = {
+                .state = SOF_DSP_PM_D3,
+        };
 	int ret;
 
+	/* set initial power state to D3 */
+	sdev->dsp_power_state = state;
+
```

Comment:
Why we got away without this before?
I'm not sure what is the issue....

Reply:
we didnt get away, it wasn't problematic because we never did multiple firmware boots while the PCI device is in D0. So for us the DSP D0 and PCI device D0 always meant the same. But if we were to boot the firmware multiple times without going through a suspend cycle, I need to put the DSP is D3 before I load & boot the firmware multiple times

Reply:
good catch. the first initialize to default power state is incorrect.